.github/workflows/build-and-test.yml

```diff
@@ -72,12 +72,17 @@ jobs:
           packages: curl xz-utils build-essential cmake m4 nasm libstdc++6
           version: 1.0
       - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@stable
         with:
-          toolchain: stable
-          override: true
-      - name: Run ffi circom tests
-        run: cd mopro-ffi && cargo test --all-features
+            toolchain: 1.81.0
+      - name: Install Nargo
+        uses: noir-lang/noirup@v0.1.3
+        with:
+            toolchain: 1.0.0-beta.1
+      - name: Run ffi tests
+        run: cd mopro-ffi && cargo test --release --all-features -- --skip test_macro_proof_zkemail
```

Comment:
Have to separate `test_macro_proof_zkemail` out, otherwise it fails.

Reply:
we used the serial test before
https://github.com/vivianjeng/mopro-zkemail-nr/blob/c338fc743eec5d2ae88dccf84281ea44a11b5e4f/src/lib.rs#L127

not sure if it solves this

Reply:
It works well on my side, fixed in [ecbe0bf](https://github.com/zkmopro/mopro/pull/394/commits/ecbe0bfa067b08b83e3f2cb702865b9ec2cab6c5)